app/controllers/clinics_controller.rb

```diff
@@ -38,59 +36,30 @@ def new
 
   # GET /clinics/1/edit
   def edit
+    @clinic = Clinic.find(params[:id])
   end
 
   # POST /clinics
   # POST /clinics.json
   def create
-    #byebug
-    is_valid = check_params(-1)
-
-    if is_valid == "None"
-        @clinic = Clinic.new(clinic_params)
+    @clinic = Clinic.new(clinic_params)
+    if @clinic.save
+      redirect_to clinics_path
     else
-        #Sets the flash[:warning] value to the rendered error message
-        flash[:warning] = is_valid
-        redirect_to new_clinic_path and return
-    end
-
-    respond_to do |format|
-      if @clinic.save
-        format.html { redirect_to @clinic, notice: 'Clinic was successfully created.' }
-        format.json { render :show, status: :created, location: @clinic }
-      else
-        format.html { render :new }
-        format.json { render json: @clinic.errors, status: :unprocessable_entity }
-      end
+      render :new
     end
   end
 
   # PATCH/PUT /clinics/1
   # PATCH/PUT /clinics/1.json
   def update
-    is_valid = check_params(@clinic.id)
-    update =  @clinic.update(clinic_params)
 
-    #byebug
-    if (is_valid == "None") and update
-        flash[:notice] = "Clinic was sucessfully updated"
+    update =  @clinic.update(clinic_params)
```

Comment:
I think you can get rid of the update variable and just call the `@clinic.update` method directly